File: AppUI_OrfDBHandler/DataListViewHandler.vb
Public Class DataListViewHandler

    Public Event LoadStart(taskTitle As String)
    Public Event LoadProgress(fractionDone As Double)
    Public Event LoadEnd()
    Public Event NumberLoadedStatus(FilteredCount As Integer, TotalCount As Integer)


    Public Sub New(listViewToFill As ListView)
        m_LVW = listViewToFill
    End Sub

    Private m_LVW As ListView

    Public Sub Load(listTable As DataTable)
        FillListView(m_LVW, listTable)
    End Sub

    Public Sub Load(listTable As DataTable, quickFilterCriteria As String)
        FillFilteredListView(m_LVW, listTable, quickFilterCriteria)
    End Sub

    Private Sub SetupPickerListView(
        lvw As ListView,
        dt As DataTable,
        Optional filterCriteria As String = "")

        Dim itemRow As DataRow
        Dim itemRows() As DataRow
        Dim filterString As String = String.Empty
        Dim proteinCount As Integer
        Dim triggerCount As Integer
        Dim counter As Integer

        If Len(filterCriteria) <> 0 Then
            filterString = "[Name] LIKE '%" & filterCriteria & "%' " &
                "OR [Description] LIKE '%" & filterCriteria & "%'"
        End If

        lvw.BeginUpdate()

        itemRows = dt.Select(filterString, "Name")
        triggerCount = itemRows.Length

        'RaiseEvent NumberLoadedStatus(itemRows.Length, dt.Rows.Count)

        proteinCount = CInt(itemRows.Length)
        If proteinCount > 20 Then
            triggerCount = CInt(proteinCount / 20)
        Else
            triggerCount = 1
        End If

        'RaiseEvent LoadStart("Filling List...")

        For Each itemRow In itemRows
            counter += 1
            Dim item As New ListViewItem
            item.Text = CStr(itemRow.Item(0))
            item.SubItems.Add(CStr(itemRow.Item(1)))
            lvw.Items.Add(item)
            'If counter Mod triggerCount > 0 Then
            '    RaiseEvent LoadProgress(CSng(counter / proteinCount))
            'End If
        Next

        lvw.EndUpdate()
    End Sub

    Protected Sub FillListView(
        listViewToFill As ListView,
        listData As DataTable)

        listViewToFill.Items.Clear()
        SetupPickerListView(listViewToFill, listData)
    End Sub

    Protected Sub FillFilteredListView(
        listViewToFill As ListView,
        listData As DataTable,
        filterString As String)

        listViewToFill.Items.Clear()
        SetupPickerListView(listViewToFill, listData, filterString)
    End Sub

End Class

File: AppUI_OrfDBHandler/DataListViewHandler.cs
using System.Data;
using System.Windows.Forms;

namespace AppUI_OrfDBHandler
{
    public class DataListViewHandler
    {
        public DataListViewHandler(ListView listViewToFill)
        {
            m_LVW = listViewToFill;
        }

        private readonly ListView m_LVW;

        public void Load(DataTable listTable)
        {
            FillListView(m_LVW, listTable);
        }

        public void Load(DataTable listTable, string quickFilterCriteria)
        {
            FillFilteredListView(m_LVW, listTable, quickFilterCriteria);
        }

        private void SetupPickerListView(
            ListView lvw,
            DataTable dt,
            string filterCriteria = "")
        {
            string filterString = string.Empty;

            if (filterCriteria.Length != 0)
            {
                filterString = "[Name] LIKE '%" + filterCriteria + "%' " +
                    "OR [Description] LIKE '%" + filterCriteria + "%'";
            }

            lvw.BeginUpdate();

            var itemRows = dt.Select(filterString, "Name");

            //NumberLoadedStatus?.Invoke(itemRows.Length, dt.Rows.Count);

            // proteinCount = Convert.ToInt32(itemRows.Length);

            //LoadStart?.Invoke("Filling List...");

            foreach (var itemRow in itemRows)
            {
                var item = new ListViewItem();
                item.Text = itemRow[0].ToString();
                item.SubItems.Add(itemRow[1].ToString());
                lvw.Items.Add(item);
            }

            lvw.EndUpdate();
        }

        protected void FillListView(
            ListView listViewToFill,
            DataTable listData)
        {
            listViewToFill.Items.Clear();
            SetupPickerListView(listViewToFill, listData);
        }

        protected void FillFilteredListView(
            ListView listViewToFill,
            DataTable listData,
            string filterString)
        {
            listViewToFill.Items.Clear();
            SetupPickerListView(listViewToFill, listData, filterString);
        }
    }
}